auth_password_settings/models/res_users.py

```diff
@@ -0,0 +1,118 @@
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+from openerp.osv import osv
+from openerp.tools.safe_eval import safe_eval
+from openerp.tools.translate import _
+import string
+
+
+class res_users(osv.osv):
+    _inherit = "res.users"
+
+    def _validate_password(self, cr, uid, password, context=None):
+        icp = self.pool.get('ir.config_parameter')
+        password_rules = []
+        config_data = {
+            'auth_password_min_character': safe_eval(
+                icp.get_param(
+                    cr,
+                    uid,
+                    'auth_password_settings.auth_password_min_character',
+                    '6'
+                )),
+            'auth_password_has_capital_letter': safe_eval(
+                icp.get_param(
+                    cr,
+                    uid,
+                    'auth_password_settings.auth_password_has_capital_letter',
+                    'False'
+                )),
+            'auth_password_has_digit': safe_eval(
+                icp.get_param(
+                    cr,
+                    uid,
+                    'auth_password_settings.auth_password_has_digit',
+                    'False'
+                )),
+            'auth_password_has_special_letter': safe_eval(
+                icp.get_param(
+                    cr,
+                    uid,
+                    'auth_password_settings.auth_password_has_special_letter',
+                    'False'
+                )),
+        }
+        password_rules.append(
+            lambda s:
+                len(s) >= config_data.get('auth_password_min_character', 6) or
+                _('Has %s or more characters') % (config_data.get(
+                    'auth_password_min_character', 6)
+                )
+        )
+
+        if (config_data.get('auth_password_has_capital_letter', False)):
+            password_rules.append(
+                lambda s: any(x.isupper() for x in s) or
+                _('Has at least One Capital letter')
+            )
+
+        if (config_data.get('auth_password_has_digit', False)):
+            password_rules.append(
+                lambda s: any(x.isdigit() for x in s) or
+                _('Has one Number')
+            )
+
+        if (config_data.get('auth_password_has_special_letter', False)):
+            password_rules.append(
+                lambda s: any(x in string.punctuation for x in s) or
+                _('Has one Special letter')
+            )
+        problems = [
+            p for p in [
+                r(password) for r in password_rules
+            ] if p and p is not True]
+        return problems
+
+    def write(self, cr, uid, ids, values, context=None):
+        if('password' in values):
+            problems = self._validate_password(
+                cr, uid, values['password'], context=context)
+            if(problems):
```

Comment:
give `_validate_password` a keyword argument indicating it should raise and raise there to have this code centralized